panels/network/panel-common.c

```diff
@@ -568,16 +568,32 @@ panel_get_dns_as_string (NMIPConfig *ip_config)
 }
 
 gchar *
-panel_get_ip6_address_as_string (NMIPConfig *ip6_config)
+panel_get_ip6_address_as_string (NMIPConfig *ip6_config, const char *what)
 {
-        GPtrArray *array;
-        NMIPAddress *address;
-
-        array = nm_ip_config_get_addresses (ip6_config);
-        if (array->len < 1)
-                return NULL;
-        address = array->pdata[0];
-        return g_strdup (nm_ip_address_get_address (address));
+        const gchar *str = NULL;
+        
+        if (!strcmp (what, "address")) {
+                g_autoptr(GPtrArray) ipv6 = NULL;
+                GPtrArray *addresses;
+
+                addresses = nm_ip_config_get_addresses (ip6_config);
+                if (addresses->len == 0 ) {
+                        goto out;
+                }
+                ipv6 = g_ptr_array_sized_new (addresses->len + 1);
+
+                for (int i = 0; i < addresses->len; i++) {
+                        g_ptr_array_add (ipv6, (char *) nm_ip_address_get_address (g_ptr_array_index (addresses, i)));
+                }
+                g_ptr_array_add (ipv6, NULL);
+                str = g_strjoinv ("\n", (char **) ipv6->pdata);
+        } else if (!strcmp (what, "gateway")) {
+                str = nm_ip_config_get_gateway (ip6_config);
+        }
+
+out:
+        return g_strdup (str);
+        //return g_strdup (nm_ip_address_get_address (address));
```

Comment:
delete?

Reply:
The comment should be deleted of course :D
The jump mark...I drafted it for consistency after the function for receiving nameservers. Of course we could directly return the value from within the if/else-clause and discard the jump.